dirsrvtests/tests/suites/cos/cos_test.py

```diff
@@ -14,9 +16,12 @@
 from lib389.idm.nscontainer import nsContainer
 from lib389.idm.user import UserAccount
 
+logging.getLogger(__name__).setLevel(logging.INFO)
+log = logging.getLogger(__name__)
+
 pytestmark = pytest.mark.tier1
 
-def test_positive(topo):
+def test_positive(topo, request):
```

Comment:
The test suite fails if I run it in DEBUGGING mode: 

```
DEBUGGING=ywa py.test -s -v dirsrvtests/tests/suites/cos/cos_test.py
```

I think it happens because the fixture should be used here instead of adding a finnalizer to the test function.
Something like this should work (feel free to modify it to your needs):

```
diff --git a/dirsrvtests/tests/suites/cos/cos_test.py b/dirsrvtests/tests/suites/cos/cos_test.py
index 16218fcc4..3cd04858b 100644
--- a/dirsrvtests/tests/suites/cos/cos_test.py
+++ b/dirsrvtests/tests/suites/cos/cos_test.py
@@ -21,7 +21,17 @@ log = logging.getLogger(__name__)
 
 pytestmark = pytest.mark.tier1
 
-def test_positive(topo, request):
+
+@pytest.fixture(scope="function")
+def reset_ignore_vattr(topo, request):
+    default_ignore_vattr_value = topo.standalone.config.get_attr_val_utf8('nsslapd-ignore-virtual-attrs')
+    def fin():
+        topo.standalone.config.set('nsslapd-ignore-virtual-attrs', default_ignore_vattr_value)
+
+    request.addfinalizer(fin)
+
+
+def test_positive(topo, reset_ignore_vattr):
     """CoS positive tests
 
     :id: a5a74235-597f-4fe8-8c38-826860927472
@@ -83,15 +93,10 @@ def test_positive(topo, request):
 
     #  CoS definition entry's cosSpecifier attribute specifies the employeeType attribute
     assert user.present('employeeType')
+    cosdef.delete()
 
-    def fin():
-        cosdef = CosClassicDefinitions(topo.standalone, DEFAULT_SUFFIX).get('cosClassicGenerateEmployeeTypeUsingnsrole')
-        cosdef.delete()
-        topo.standalone.config.set('nsslapd-ignore-virtual-attrs', 'on')
 
-    request.addfinalizer(fin)
-
-def test_vattr_on_cos_definition(topo, request):
+def test_vattr_on_cos_definition(topo, reset_ignore_vattr):
     """Test nsslapd-ignore-virtual-attrs configuration attribute
        The attribute is ON by default. If a cos definition is
        added it is moved to OFF
@@ -124,7 +129,7 @@ def test_vattr_on_cos_definition(topo, request):
                   'cosAttribute': 'employeeType',
                   'cosSpecifier': 'nsrole',
                   'cn': 'cosClassicGenerateEmployeeTypeUsingnsrole'}
-    CosClassicDefinition(topo.standalone,'cn=cosClassicGenerateEmployeeTypeUsingnsrole,{}'.format(DEFAULT_SUFFIX))\
+    cosdef = CosClassicDefinition(topo.standalone,'cn=cosClassicGenerateEmployeeTypeUsingnsrole,{}'.format(DEFAULT_SUFFIX))\
         .create(properties=properties)
 
     log.info("Check the default value of attribute nsslapd-ignore-virtual-attrs should be OFF")
@@ -133,17 +138,9 @@ def test_vattr_on_cos_definition(topo, request):
 
     topo.standalone.stop()
     assert topo.standalone.searchErrorsLog("slapi_vattrspi_regattr - Because employeeType is a new registered virtual attribute , nsslapd-ignore-virtual-attrs set to \'off\'")
+    topo.standalone.start()
+    cosdef.delete()
 
-    def fin():
-        topo.standalone.restart()
-        try:
-            cosdef = CosClassicDefinitions(topo.standalone, DEFAULT_SUFFIX).get('cosClassicGenerateEmployeeTypeUsingnsrole')
-            cosdef.delete()
-        except:
-            pass
-        topo.standalone.config.set('nsslapd-ignore-virtual-attrs', 'on')
-
-    request.addfinalizer(fin)
 
 if __name__ == "__main__":
     CURRENT_FILE = os.path.realpath(__file__)
```